### PR TITLE
[core] Add time to TableSchema

### LINKED ISSUE #1715: [Feature] Add creation time to TableSchema
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

User would like to be able to know the time of schema changes.
I have a supporting management system that needs to display the schema change records of tables. time are required so that users can know when the schema was modified and what changes were made.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by shidayang**
### Purpose
Linked issue: close #1715

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Also modify `SchemasTable`?

**Comment by shidayang** [comment]
> Also modify `SchemasTable`?
I have modified the SchemasTable. cc

File: docs/content/how-to/system-tables.md [hunk 1]
```diff
@@ -65,13 +65,13 @@ You can query the historical schemas of the table through schemas table.
...
 /*
-+-----------+--------------------------------+----------------+--------------+---------+---------+
-| schema_id |                         fields | partition_keys | primary_keys | options | comment |
-+-----------+--------------------------------+----------------+--------------+---------+---------+
-|         0 | [{"id":0,"name":"word","typ... |             [] |     ["word"] |      {} |         |
-|         1 | [{"id":0,"name":"word","typ... |             [] |     ["word"] |      {} |         |
-|         2 | [{"id":0,"name":"word","typ... |             [] |     ["word"] |      {} |         |
-+-----------+--------------------------------+----------------+--------------+---------+---------+
++-----------+--------------------------------+----------------+--------------+---------+---------+-------------------------+
+| schema_id |                         fields | partition_keys | primary_keys | options | comment |      effective_time     |
```
Comment (JingsongLi) [inline docs/content/how-to/system-tables.md:69]: Why using `effective_time` instead of update_time?

Comment (shidayang) [reply docs/content/how-to/system-tables.md:69]: Upon reflection, "update_time" does indeed sound better.

**Comment by shidayang** [comment]
The CI reported that the MySqlSyncDatabaseTableListITCase.testActionRunResult unit test failed, which should not be related to this PR. It is likely that there is an issue with the unit test itself. Can you trigger the CI again? @JingsongLi

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**